main/main.go

```diff
@@ -243,35 +243,45 @@ func runReplicate(
 
 	// Signal Handling
 	go func() {
+		stoppedProfiling := false
+
 		for {
 			sig := <-sigchan
 			log.Infof("Got a signal %d (%s)", sig, sig.String())
 
-			// If it's a SIGUSR1 and the memprofile flag was set, then dump a pprof and continue running
-			// https://golang.org/pkg/runtime/pprof/
+			// If it's a SIGUSR1 and the cpuprofile/memprofile flag was set, then dump pprof files and continue running
+			// https://golang.org/pkg/runtime/pprof
 			if sig == syscall.SIGUSR1 {
 
+				if stoppedProfiling != true {
+					if cpuprofile != "" {
+						log.Warn("Stopping CPU profiling")
+						pprof.StopCPUProfile()
+						log.Infof("Wrote pprof cpu profile to: '%s'", cpuprofile)
+					}
+
+					if memprofile != "" {
+						time.Sleep(1 * time.Second)
+						memProfile(fmt.Sprintf(memprofile))
+					}
+
+					stoppedProfiling = true
+				}
+
+				continue
```

Comment:
I think having this `continue` here means that you can't send two `SIGUSR1` to start/stop. Can you document the code a bit more. Unfortunately we made this a little convoluted.

Reply:
Correct. Done, see latest commit.